hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -340,10 +332,24 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to reconcile hostedcontrolplane: %w", err)
 	}
 
+	// Reconcile external AWSCluster
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(hcp), hcp); err != nil {
+		r.Log.Error(err, "failed to get control plane ref")
+		return reconcile.Result{}, err
+	}
+
+	awsCluster := controlplaneoperator.ExternalInfraCluster(controlPlaneNamespace.Name, hcluster.Name)
```

Comment:
should this only happen when Platform.Type == "AWS" ? Also s/ExternalInfraCluster/AWSCluster/
do we still create/reconcile an AWSCluster when Platform.Type == "None"?

Reply:
>should this only happen when Platform.Type == "AWS
yes.
>do we still create/reconcile an AWSCluster when Platform.Type == "None"?
No.

Each provider must have their own infraCluster CR.

Let me update.